Source/ACE/Factories/SpellObjectFactory.cs

```diff
@@ -0,0 +1,84 @@
+﻿using ACE.Entity;
+using ACE.Entity.Enum;
+using ACE.Network.Enum;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ACE.Factories
+{
+
+    /// <summary>
+    /// Spell Factory for creating objects related to Spells.
+    /// </summary>
+    public class SpellObjectFactory
+    {
+        public static WorldObject CreateSpell(uint templateId, Position position, AceVector3 velocity, float friction, float electicity)
+        {
+
+            ushort weenieClassId = 0;
+            MutableWorldObject mobj = new MutableWorldObject(ObjectType.MissileWeapon, new ObjectGuid(CommonObjectFactory.DynamicObjectId, GuidType.None), "Spell", weenieClassId, ObjectDescriptionFlag.None, WeenieHeaderFlag.Spell, position);
```

Comment:
pull your switch block above this to find out what weenieClassId you need and pass it in.  We discussed this several times.